### DISCUSSION TITLE
Portable to multi-arch installer migration

### DISCUSSION THREAD

**Discussion by NickeManarin** [chocolatey/choco#2494] [Q&A] [answered]
So, right now I'm distributing my software as a zip (just an executable inside) that is downloaded from GitHub.
I want to convert the package to add support for x64, x86 and arm64 installers.

What's the migration path for my package?

I noticed that chocolatey does not support arm64. Is that right?

**Comment by pauby** [comment, marked as answer]
Chocolatey CLI doesn't support ARM64 at this time, no.

**Reply by NickeManarin** [reply]
Ok, so I can create a package with the x64 and x86 installer. But since I have been shipping previous versions as a ZIP that gets uncompressed, I need to delete that package before installing the new version.

**Reply by TheCakeIsNaOH** [reply]
> But since I have been shipping previous versions as a ZIP that gets uncompressed, I need to delete that package before installing the new version.

Are you talking about a package on the community repository, or in an internal repository?

You should be able to move to `.install` and `.portable` packages:
https://docs.chocolatey.org/en-us/faqs#what-is-the-difference-between-packages-no-suffix-as-compared-to.install.portable

The `.install` package would contain the installers, while the `.portable` version would contain the archives.

**Reply by NickeManarin** [reply]
@TheCakeIsNaOH This package: https://community.chocolatey.org/packages/screentogif

**Reply by NickeManarin** [reply]
I think that I'll create another package ending with `.install` (with x86 and x64) and update the current one, adding the (x64 variants).
Since arm64 is not currently supported, I'll leave for later.

**Reply by TheCakeIsNaOH** [reply]
Ok, so yes, creating a `.install` package would be the best option. That would entail creating a `screentogif.install` package which downloads and runs the installers. After that is completed, duplicate the `screentogif` package renamed as `screentogif.portable`. Finally, make the `screentogif` package into a metapackage by removing the `.ps1` scripts and taking a dependency on either the install or portable packages.

See the `7zip`, `7zip.install` and `7zip.portable` packages as an example: 
https://github.com/chocolatey-community/chocolatey-packages/tree/master/automatic

If you want more examples, let me know.

**Reply by NickeManarin** [reply]
@TheCakeIsNaOH Done! Just waiting for approval.
Thank you!

https://community.chocolatey.org/packages/screentogif/2.35.2
https://community.chocolatey.org/packages/screentogif.install/2.35.2
https://community.chocolatey.org/packages/screentogif.portable/2.35.2
